cql3/expr/expression.cc

```diff
@@ -0,0 +1,872 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "expression.hh"
+
+#include <boost/algorithm/cxx11/all_of.hpp>
+#include <boost/algorithm/cxx11/any_of.hpp>
+#include <boost/range/adaptors.hpp>
+
+#include "cql3/lists.hh"
+#include "cql3/tuples.hh"
+#include "types/list.hh"
+#include "types/map.hh"
+#include "types/set.hh"
+#include "utils/like_matcher.hh"
+
+namespace cql3 {
+namespace expr {
```

Comment:
Note (no need to change): you can write `namespace cql3::expr`, reducing some repetitions.

Reply:
Note2 (ditto): I expect we'll move this out of the cql3 namespace eventually and make it a common resource for all front-ends.